### DISCUSSION TITLE
React-hook-form validation

### DISCUSSION THREAD

**Discussion by kshitiz91** [react-hook-form/react-hook-form#8270] [Q&A]
Is there any side effect method or callback which can be invoked after the validation of input is executed (pass/fail) in case when mode is set to onBlur . Callback need to be specific to each input in form

**Comment by Moshyfawn** [comment]
You can pass an `onBlur` callback to your form element.

```jsx
<form onBlur={() => handleBlur} />
```

**Reply by kshitiz91** [reply]
Sorry, I want it on each input element and I already have onBlur event which is running before the validation of field take place.

**Reply by Moshyfawn** [reply]
I'm not sure I understand your use-case. If you want to do stuff `onBlur` element, you can pass the callback to `register`
```jsx
<input {...register("fieldName", { onBlur: handleBlur })} />
```

With controlled inputs, just pass your `onBlur` logic into the input's `onBlur`

**Reply by Moshyfawn** [reply]
BTW, `onBlur` on the `form` element will fire on the form's input's blur, not the form element.

**Reply by kshitiz91** [reply]
Yeah, onBlur handler doesn't get the updated validation of the input. validation happens after the handler call is completed.
